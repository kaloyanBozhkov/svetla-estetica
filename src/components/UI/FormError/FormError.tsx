// import common types
import type { ReactElement } from 'react'
import type { FormErrorType } from 'common/types'

// import atom styles
import styles from './styles.module.scss'

export default function FormError({ error }: { error: FormErrorType }): ReactElement | null {
    return error && typeof error === 'string' ? <p className={styles.errorMsg}>{error}</p> : null
}
