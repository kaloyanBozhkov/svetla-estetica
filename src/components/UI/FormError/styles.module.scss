@import "scss/mixins.scss";
@import "scss/variables.scss";

.errorMsg {
    @include animate(popIn, 300ms);

    color: $pink;
}
