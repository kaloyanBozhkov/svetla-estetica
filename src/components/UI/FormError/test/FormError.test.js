import { shallow } from 'enzyme'
import React from 'react'

import FormError from '../FormError'

describe('testing top level FormError', () => {
    it('should render component if valid error str provided', () => {
        expect(shallow(<FormError error="some error" />)).toMatchSnapshot()
    })

    it('should not render anything if error is obj or falsy', () => {
        expect(shallow(<FormError error={{ field: 'error' }} />).find('.errorMsg').length).toBe(0)
        expect(shallow(<FormError />).find('.errorMsg').length).toBe(0)
    })
})
