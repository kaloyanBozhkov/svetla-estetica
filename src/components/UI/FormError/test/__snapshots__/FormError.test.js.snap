// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`testing top level FormError should render component if valid error str provided 1`] = `
<p
  className="errorMsg"
>
  some error
</p>
`;
